.github/workflows/ci-cd.yml

```diff
@@ -127,12 +124,36 @@ jobs:
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  release:
+    needs:
+      - build_release
+
+    runs-on: ubuntu-latest
+    environment: pypi
+    concurrency: release
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
       - name: Publish package distributions to PyPI
         uses: pypa/gh-action-pypi-publish@release/v1
-        if: steps.release.outputs.released == 'true'
+        if: steps.build_release.outputs.released == 'true'
 
       - name: Publish package distributions to GitHub Releases
         uses: python-semantic-release/upload-to-gh-release@main
```

Comment:
Note that using moving pointers like `@main` can contribute to instability in the CI and be a supply chain attack surface. It's best to use versioned pointers. In the nutshell, actions are like normal software dependencies.

Reply:
It looks like upstream doesn't version so I pinned it to `1eeda76de33f20e9341a93b0e697c79773c6914c`